src/api/firebase.js

```diff
@@ -179,21 +181,63 @@ export async function addItem(listPath, { itemName, daysUntilNextPurchase }) {
 		dateCreated: new Date(),
 		// NOTE: This is null because the item has just been created.
 		// We'll use updateItem to put a Date here when the item is purchased!
-		dateLastPurchased: null,
+		dateLastPurchased: [new Date()],
 		dateNextPurchased: getFutureDate(daysUntilNextPurchase),
 		name: itemName,
 		totalPurchases: 0,
 	});
+	console.log(newItem);
 	return newItem;
 }
 
+//Helper function to handle estimation logic
+async function handleCalculateEstimate(listRef) {
+	const selectedItem = await getDoc(listRef);
+
+	//access array of dateLastPurchased for the selected item
```

Comment:
praise: really like the comments you added here; there's a lot going on so they're very helpful for tracking the process!